.github/workflows/build-and-upload.yml

```
@@ -0,0 +1,200 @@
+  ECR_OPERATOR_STAGING_REPO: ${{ vars.ECR_OPERATOR_STAGING_REPO }}
+  ECR_OPERATOR_RELEASE_IMAGE: ${{ vars.ECR_OPERATOR_RELEASE_IMAGE }}
+
+on:
+  workflow_call:
+    inputs:
+      release:
+        description: 'Release Artifact'
+        required: false
+        default: false
+        type: boolean
+      e2e-test-bypass-link:
+        description: 'Failed E2E Test Run Link (for bypass)'
+        required: false
+        type: string
+      e2e-test-bypass-approver:
+        description: 'Approver GitHub Username (for bypass)'
+        required: false
+        type: string
+      tag:
+        description: 'Staging Artifact Tag'
+        required: false
+        default: 'integration'
+        type: string
+      target-sha:
+        default: ${{ github.sha}}
+        type: string
+      repository:
+        default: ${{ github.repository }}
+        type: string
+      test-image-before-upload:
+        description: "Run Test on the new container image"
+        default: true
+        type: boolean
+      region: 
+        description: "AWS Region to run the build/test on"
+        default: "us-west-2"
+        type: string
+
+
+jobs:
+  MakeBinary:
+    name: 'MakeContainerImage'
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ inputs.target-sha }}    
+          repository: ${{inputs.repository}}
+
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v4
+        with:
+          go-version: ~1.19.6
+          cache: false
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ env.AWS_ASSUME_ROLE }}
+          aws-region: ${{inputs.region}}
+
+      - name: Login to ECR
+        if: steps.cached_binaries.outputs.cache-hit == false
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Set up Docker Buildx
+        if: steps.cached_binaries.outputs.cache-hit == false
+        uses: docker/setup-buildx-action@v1
+
+      - name: Set up QEMU
+        if: steps.cached_binaries.outputs.cache-hit == false
+        uses: docker/setup-qemu-action@v1
+
+      - name: Build Cloudwatch Agent Operator Image and push to ECR
+        uses: docker/build-push-action@v4
+        if: steps.cached_binaries.outputs.cache-hit == false
+        with:
+          file: ./Dockerfile
+          context: .
+          push: true
+          platforms: linux/amd64, linux/arm64
+          tags: |
+           ${{ env.ECR_OPERATOR_STAGING_REPO }}:${{ inputs.tag }}
+           ${{ env.ECR_OPERATOR_STAGING_REPO }}:${{ inputs.target-sha }}
+  MakeTABinary:
+    name: 'MakeTargetAllocatorImage'
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

Make sure we checkout to the right repo.
```suggestion
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
          ref: ${{ inputs.target-sha }}
          repository: ${{inputs.repository}}
```